components/org.wso2.carbon.identity.user.endpoint/src/main/java/org/wso2/carbon/identity/user/endpoint/impl/ResendCodeApiServiceImpl.java

```diff
@@ -0,0 +1,86 @@
+ */
+
+package org.wso2.carbon.identity.user.endpoint.impl;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.wso2.carbon.identity.recovery.IdentityRecoveryClientException;
+import org.wso2.carbon.identity.recovery.IdentityRecoveryConstants;
+import org.wso2.carbon.identity.recovery.IdentityRecoveryRuntimeException;
+import org.wso2.carbon.identity.recovery.IdentityRecoveryServerException;
+import org.wso2.carbon.identity.recovery.bean.NotificationResponseBean;
+import org.wso2.carbon.identity.recovery.signup.UserSelfSignUpManager;
+import org.wso2.carbon.identity.user.endpoint.NotFoundException;
+import org.wso2.carbon.identity.user.endpoint.ResendCodeApiService;
+import org.wso2.carbon.identity.user.endpoint.dto.ErrorDTO;
+import org.wso2.carbon.identity.user.endpoint.dto.ResendCodeRequestDTO;
+import org.wso2.carbon.identity.user.endpoint.dto.ResendConfirmationCodeResponseDTO;
+import org.wso2.carbon.identity.user.endpoint.dto.UserDTO;
+import org.wso2.carbon.identity.user.endpoint.util.Utils;
+
+import javax.ws.rs.core.Response;
+
+/**
+ * Micro service implementation of /resendcode endpoint.
+ */
+public class ResendCodeApiServiceImpl extends ResendCodeApiService {
+
+    private static final Logger log = LoggerFactory.getLogger(ResendCodeApiServiceImpl.class);
+
+    @Override
+    public Response resendCodePost(ResendCodeRequestDTO user) throws NotFoundException {
+
+        UserSelfSignUpManager userSelfSignUpManager = Utils.getUserSelfSignUpManager();
+        UserDTO userDTO = user.getUser();
+
+        NotificationResponseBean notificationResponseBean;
+        try {
+
+            notificationResponseBean = userSelfSignUpManager.resendConfirmationCode(Utils.getUsernameClaim(
+                    userDTO.getUsername()), userDTO.getDomain(), Utils.getProperties(user.getProperties()));
+        } catch (IdentityRecoveryClientException e) {
+
+            if (log.isDebugEnabled()) {
+                log.debug("Client error while registering self sign-up user: " + user.getUser().getUsername(), e);
+            }
+            ErrorDTO errorDTO = Utils.buildBadRequestErrorDTO(e.getErrorCode(), e.getErrorDescription());
+            return Response.status(Response.Status.BAD_REQUEST).entity(errorDTO).build();
+        } catch (IdentityRecoveryServerException e) {
+
+            log.error("Server error while user confirmation.", e);
+            ErrorDTO errorDTO = Utils.buildInternalServerErrorDTO(e.getErrorCode(), e.getErrorDescription());
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR).entity(errorDTO).build();
+        } catch (IdentityRecoveryRuntimeException e) {
+
+            log.error("Server error while user confirmation.", e);
+            ErrorDTO errorDTO = Utils.buildInternalServerErrorDTO(e.getErrorCode(), e.getMessage());
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR).entity(errorDTO).build();
+        } catch (Throwable e) {
```

Comment:
Any specific reason to catch Throwable, if so, please add it as a code comment. let the MSF4J to handle it otherwise.